tests/testsuite/publish.rs

```diff
@@ -130,6 +130,54 @@ You may press ctrl-c to skip waiting; the crate should be available shortly.
     validate_upload_foo();
 }
 
+#[cargo_test]
+fn duplicate_version() {
+    let arc: Arc<Mutex<u32>> = Arc::new(Mutex::new(0));
+    let registry_dupl = RegistryBuilder::new()
+        .http_api()
+        .http_index()
+        .add_responder("/index/3/f/foo", move |req, server| {
+            let mut lock = arc.lock().unwrap();
+            *lock += 1;
+            if *lock <= 1 {
+                server.not_found(req)
+            } else {
+                server.index(req)
+            }
+        })
+        .build();
```

Comment:
I can't remember, was there a reason we couldn't publish the package like we normally do in tests?

Reply:
This was initially added because publish during tests did not have any memory of previously uploaded packages by default. Not sure if that changed, but test-publish works fine without this responder now.

Reply:
And it's updated.